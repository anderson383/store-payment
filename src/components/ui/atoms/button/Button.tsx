import { IconLoader2 } from '@tabler/icons-react';
import styles from './Button.module.scss'
interface ButtonProps {
  text: string
  onClick?: () => void;
  variant?: 'outlined' | 'simple' | 'rounded' | 'rounded_outlined';
  color?: 'primary' | 'secondary'
  size?: 'fullwidth' | 'md' | 'sm'
  type?: 'button' | 'submit' | 'reset';
  icon?: React.ReactNode;
  loading?: boolean;
  disabled?: boolean;
}

export const Button:React.FC<ButtonProps> = ({
  type = 'button',
  text,
  variant = ' simple',
  color = 'primary',
  size = 'md',
  icon,
  loading =false,
  disabled = false,
  onClick = () => {} }
) => {
  return (
    <button
      role='button'
      disabled={disabled || loading}
      type={type}
      onClick={onClick}
      className={
        `
        ${styles.button}
        ${styles['button__' +variant]}
        ${styles['button__' +size]}
        ${styles['button__' + color]}
        ${icon ? styles.button_icon : ''}
        ${loading ? styles.loading : ''}
        `
      }
    >
      {text}
      {
        icon && loading ? (
          <IconLoader2 />
        ): icon
      }
    </button>
  )
}