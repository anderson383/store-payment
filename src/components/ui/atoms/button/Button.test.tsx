
import {
  fireEvent,
  render,
  screen
} from '@testing-library/react';
import { BrowserRouter } from 'react-router-dom';
import { Button } from './Button';
import { act } from 'react';

const onClickMock = jest.fn()


describe('Tests in Button component', () => {
  const setup = () => {
    return render(
      <BrowserRouter>
        <Button text='Test Button' color='primary' onClick={onClickMock}   />
      </BrowserRouter>
    );
  };

  test('Should render the component', async () => {
    const {asFragment} = setup();
    
    await act(() => Promise.resolve())
    expect(asFragment()).toMatchSnapshot();
  });

  test('Should click button', async () => {
    // Arrange
    const {asFragment} = setup();
    
    const button = screen.getByText('Test Button');

    fireEvent.click(button)
    
    // Assert
    expect(onClickMock).toBeCalled();
  });
});
