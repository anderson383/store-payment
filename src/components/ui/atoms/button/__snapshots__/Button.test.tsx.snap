// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests in Button component Should render the component 1`] = `
<DocumentFragment>
  <button
    class="
        button
        button__ simple
        button__md
        button__primary
        
        
        "
    role="button"
    type="button"
  >
    Test Button
  </button>
</DocumentFragment>
`;
